NetCord/Gateway/GatewayClient.cs

```diff
@@ -17,71 +20,385 @@ public partial class GatewayClient : WebSocketClient, IEntity
+
+    /// <summary>
+    /// Sent when a message is created. Ephemeral messages do not use the guild channel. Because of this, they are tied to the <see cref="GatewayIntents.DirectMessages"/> intent, and the message object won't include a <see cref="Message.GuildId"/> or <c>member</c>.
+    /// The inner payload is a message object with set <see cref="Message.GuildId"/>, <c>member</c> and <c>mentions</c> fields.
+    /// </summary>
     public event Func<Message, ValueTask>? MessageCreate;
+
+    /// <summary>
+    /// Sent when a message is updated. Ephemeral messages do not use the guild channel. Because of this, they are tied to the <see cref="GatewayIntents.DirectMessages"/> intent, and the message object won't include a <see cref="Message.GuildId"/> or <c>member</c>.
+    /// The inner payload is a message object with the same extra fields as <see cref="MessageCreate"/>. Updates may contain only a partial <see cref="Message"/> object (but will always contain an <c>ID</c> and <c>channel_id</c>).
+    /// </summary>
     public event Func<Message, ValueTask>? MessageUpdate;
+
+    /// <summary>
+    /// Sent when a message is deleted. Ephemeral messages do not use the guild channel. Because of this, they are tied to the <see cref="GatewayIntents.DirectMessages"/> intent, and the message object won't include a <see cref="Message.GuildId"/> or <c>member</c>.
+    /// </summary>
     public event Func<MessageDeleteEventArgs, ValueTask>? MessageDelete;
+
+    /// <summary>
+    /// Sent when multiple messages are deleted at once.
+    /// </summary>
     public event Func<MessageDeleteBulkEventArgs, ValueTask>? MessageDeleteBulk;
+
+    /// <summary>
+    /// Sent when a user adds a reaction to a message.
+    /// </summary>
     public event Func<MessageReactionAddEventArgs, ValueTask>? MessageReactionAdd;
+
+    /// <summary>
+    /// Sent when a user removes a reaction from a message.
+    /// </summary>
     public event Func<MessageReactionRemoveEventArgs, ValueTask>? MessageReactionRemove;
+
+    /// <summary>
+    /// Sent when a user explicitly removes all reactions from a message.
+    /// </summary>
     public event Func<MessageReactionRemoveAllEventArgs, ValueTask>? MessageReactionRemoveAll;
+
+    /// <summary>
+    /// Sent when a bot removes all instances of a given emoji from the reactions of a message.
+    /// </summary>
     public event Func<MessageReactionRemoveEmojiEventArgs, ValueTask>? MessageReactionRemoveEmoji;
+
+    /// <summary>
+    /// Sent when a user's presence or info, such as their name or avatar, is updated. Requires the <see cref="GatewayIntents.GuildPresences"/> intent.
+    /// The user object within this event can be partial, with the id being the only required field, everything else is optional. Along with this limitation, no fields are required, and the types of the fields are <b>not validated</b>. You should expect <b>any</b> combination of fields and types within this event.
```

Comment:
I don't think we should document so many details that are not even usable for a user. Also I am surprised that the event is sent for example when a user name or avatar is updated rather than a `GuildUserUpdate` event.

Reply:
This entire event sounds disastrous, so I'm not sure how else I could go about documenting it.